analysis/coverage_data_utils.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility functions for coverage data calculation."""
+
+import posixpath
+from collections import defaultdict
```

Comment:
nit: just import collections and do `collections.defaultdict`.
We don't always obey this convention (though 99% of the time we do), but in the styleguide it says not to import functions or classes from modules and to import the entire module instead.